### PR TITLE
Send event on referencing large file

### PR DESCRIPTION
**Pull request by sheetalkamat**
Event when referencing large file as per https://github.com/Microsoft/TypeScript/pull/26169#issuecomment-410157807

### CODE REVIEW

**Comment by amcasey** [comment]
If this were turning into telemetry (which I don't believe it is), the file size would be okay but the file path would not.

**Comment by amcasey** [comment]
@armanio123, should VS respond to this event?  Do we want to show an info message or something?  This seems similar to LS disabled.

**Comment by armanio123** [comment]
I think it might be a good idea to show an info message, just to make the user aware of what's happening on his project. Nevertheless, I don't think the feedback we have received relates to this scenario specifically or if we can get an action from the user by showing the message.

**Comment by sheetalkamat** [comment]
Is this good to merge ?

**Comment by amcasey** [comment]
I have no objections, but Ryan only added me by accident.  I know @sandersn is adding an event, so he might have thoughts.
As I mentioned above, as long as this doesn't turn into telemetry, I have no GDPR objections.

**Comment by sheetalkamat** [comment]
@RyanCavanaugh @sandersn Ok to merge ?

**Comment by sandersn** [comment]
I’m interested in this PR as an example of how to add a new flag, but I’ll take a look at the other parts now.

File: src/server/protocol.ts [hunk 1]
```diff
@@ -2436,6 +2436,27 @@ namespace ts.server.protocol {
         openFiles: string[];
     }
 
+    export type LargeFileReferencedEventName = "largeFileReferenced";
```
Comment (sandersn) [inline src/server/protocol.ts]: Why is this code duplicated? I guess that’s one public and the other private, but why aren’t they shared?

Comment (sheetalkamat) [reply src/server/protocol.ts]: That's because we want to be able to build protocol.ts separately as a standalone to protocol.d.ts which editors consume.

**Pull request merged**